File: C/C_DSA_QS_EASY/PROB_3386_Button_with_Longest_Push_Time.c
#include "stdio.h"
#include "limits.h"

int buttonWithLongestTime(int** events, int eventsSize, int* eventsColSize) {
    int prev_time = 0;
    int max_time = INT_MIN;
    int idx = 0;
    for(int event = 0 ; event < eventsSize ; event++){
        int time = events[event][1] - prev_time;
        prev_time = events[event][1];

        if(time > max_time){
            max_time = time;
            idx = events[event][0];
        }else if(time == max_time && events[event][0] < idx){
            idx = events[event][0];
        }
    }
    return idx;
}

File: PYTHON/EASY_PY_LEETCODE/PROB_3386_Button_with_Longest_Push_Time.py
from typing import *
class Solution:
    def buttonWithLongestTime(self, events: List[List[int]]) -> int:
        prev_time = 0
        _max_time = float('-inf')
        _idx = 0
        for event in events:
            _time = event[1] - prev_time
            prev_time = event[1]
            if _time > _max_time:
                _max_time = _time
                _idx = event[0]
            elif _time == _max_time and event[0] < _idx:
                    _idx = event[0]
        return _idx
